src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateBuilder.java

```diff
@@ -644,4 +649,46 @@ private List<Long> parseSupplementalGroupList(String gids) {
         }
         return Collections.unmodifiableList(builder);
     }
+
+    /**
+     * <p>
+     * {@link PodDecorator} allowing to inject in {@code jnlp} containers definition a {@code securityContext} definition allowing to use the
+     * {@code restricted} <a href="https://kubernetes.io/docs/concepts/security/pod-security-standards/">Pod Security Standard</a>.
+     * </p>
+     * <p>
+     * See <a href="https://issues.jenkins.io/browse/JENKINS-71639">JENKINS-71639</a> for more details.
+     * </p>
+     */
+    @Extension
+    public static class RestrictedPssSecurityContextInjector implements PodDecorator {
+
+        @NonNull
+        @Override
+        public Pod decorate(@NonNull KubernetesCloud kubernetesCloud, @NonNull Pod pod) {
+            if (kubernetesCloud.isRestrictedPssSecurityContext()) {
+                Optional<Container> maybeJNLP = pod.getSpec().getContainers().stream().filter(container -> JNLP_NAME.equals(container.getName())).findFirst();
+
+                maybeJNLP.ifPresentOrElse(jnlp -> {
+                    if (jnlp.getSecurityContext() != null) {
+                        LOGGER.warning(() -> "Overriding the existing JNLP container Security Context due to the configured restricted PSP injection");
```

Comment:
This is going to go into the global log and not be actionable at all.

Anyway I think this should be merging in the specified changes, rather than clobbering unrelated `securityContext` fields like specifying an FS group or whatever.

Reply:
Oh that is a nice idea indeed.

I changed the code to update the existing `securityContext` in case one is found.